index.bs

```diff
@@ -2415,6 +2567,12 @@ To <dfn noexport>parse source-registration JSON</dfn> given a [=byte sequence=]
     :: |debugCookieSet|
     : [=attribution source/fenced=]
     :: |fenced|
+    : [=attribution source/attribution scopes=]
+    :: |attributionScopes|
```

Comment:
There is a type mismatch here, as `parse attribution scopes` can return null (currently, though I don't think it needs to), but this field is not nullable.

Reply:
As discussed in a different thread, triggers with empty scopes will match none of the sources. If we would reserve it for special meaning, we are essentially introducing a new attribution scope (= null), and adtech can achieve that by increasing `attribution scope limit`.

Reply:
Nice catch, updated the [=attribution source/attribution scopes=] to be non null.

Reply:
This hasn't been fixed: `parse attribution scopes` can return null but the field is not nullable. Maybe you want to have a step after line 2462 like

```
If |attributionScopes| is null,
  set |attributionScopes| to a new [=set=].
```
?

Reply:
Done. we can simply early return empty set.